crates/miden-objects/src/errors.rs

```diff
@@ -324,8 +324,23 @@ pub enum AssetError {
       expected_ty = AccountType::NonFungibleFaucet
     )]
     NonFungibleFaucetIdTypeMismatch(AccountIdPrefix),
-    #[error("{0}")]
-    TokenSymbolError(String),
+    #[error("provided token symbol invalid")]
+    TokenSymbolError(#[source] TokenSymbolError),
+}
+
+// TOKEN SYMBOL ERROR
+// ================================================================================================
+
+#[derive(Debug, Error)]
+pub enum TokenSymbolError {
+    #[error("token symbol value {0} cannot exceed {1}")]
+    ValueTooLarge(u64, u64),
+    #[error("token symbol should have length between 1 and 6 characters, but {0} was provided")]
+    InvalidLength(usize),
+    #[error("token symbol {0} contains characters that are not uppercase ASCII")]
```

Comment:
```suggestion
    #[error("token symbol `{0}` contains characters that are not uppercase ASCII")]
```

Nit: I would put user input into backticks.